actors/builtin/miner/miner_actor.go

```diff
@@ -2144,6 +2135,22 @@ func (a Actor) ProveReplicaUpdates(rt Runtime, params *ProveReplicaUpdatesParams
 			continue
 		}
 
+		if update.Deadline >= WPoStPeriodDeadlines {
+			rt.Log(rtt.INFO, "deadline %d not in range 0..%d, skipping sector %d", update.Deadline, WPoStPeriodDeadlines, update.SectorID)
+			continue
+		}
+
+		if !update.NewSealedSectorCID.Defined() {
+			rt.Abortf(exitcode.ErrIllegalArgument, "new sealed CID undefined")
+		}
+
+		if update.NewSealedSectorCID.Prefix() != SealedCIDPrefix {
+			rt.Abortf(exitcode.ErrIllegalArgument, "new sealed CID had wrong prefix")
+		}
+
+		err = stReadOnly.CheckSectorHealth(store, update.Deadline, update.Partition, update.SectorID)
```

Comment:
A subtle problem with this is that it misses unproven sectors in this check.  These will be disallowed later on when calling ReplaceSectors so this method isn't broken but you want to do this here too for error handling reasons.

Reply:
You'll want to make a new check sector health method extended to check for unproven sectors

Reply:
Nice catch, thanks!